### ISSUE TITLE
Adding flexibility to FD calculations

### ISSUE THREAD

**Issue by skeyser** [flaviomoc/DMSD#6]
Thank you for developing a neat package! I have been using phyloraster for estimating PD, but was curious about using this package for corresponding analyses with FD. My "issue" is more of a question.

I can't tell from the vignette or corresponding documentation but I wondering if there is a way to incorporate some flexibility into the FD estimation? For example, if I wanted to add trait weights or use a gower distance matrix for mixed data types is this possible with divraster? Also, would it be possible or easier to have an option for users to supply their own functional dendrogram?

**Comment by skeyser** [comment]
I was digging in the source code and realized the fix *could* be simple from the users perspective. I was able to take my functional dendrogram (object class 'hclust') and turn it into a "phylo" object with ape::as.phylo. If I am correct, it looks like the underlying diversity function in spat.alpha is BAT::alpha() which is supplied the community and a tree (either 'hclust' or 'phylo'). The 'hclust' object branch lengths are scaled via branch.len/2 per ape::as.phylo documentation, but the underlying relationship between the BAT::alpha() run between the 'hclust' and 'phylo' object is 1:1 (I added a figure). I tested this using 10000 randomly generated communities from my own data of interest (i.e., the species and traits are real but the presence/absence data are just a sample from a random binomial process). Running BAT::alpha() with the original 'hclust' and the new hclust to phylo object are identical (I needed to multiply branch lengths by a scaler of 2 to get identical lengths). Using this procedure the user-defined functional dendrogram can be slipped into the alpha diversity function. Not entirely sure if my thinking is off-base but seems to work for now. 
![Test_FD_Divraster](https://github.com/flaviomoc/divraster/assets/35746882/0135c639-5a68-42d5-b649-0a7526a6961f)
